### PR TITLE
Find the open port when running MultiProcessExecutor.

### PR DESCRIPTION
**Pull request by yhwen**

### CODE REVIEW

File: nvflare/app_common/pt/pt_multi_process_executor.py [hunk 1]
```diff
@@ -26,5 +27,14 @@ def get_multi_process_command(self) -> str:
...
             + " --nnodes=1 --node_rank=0"
-            + ' --master_addr="localhost" --master_port=1234'
+            + ' --master_addr="localhost" --master_port='
+            + str(self._get_open_port())
         )
+
+    def _get_open_port(self):
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.bind(("", 0))
+        s.listen(1)
```
Comment (IsaacYangSLA) [inline nvflare/app_common/pt/pt_multi_process_executor.py:37]: Not sure if listen is necessary.  The kernel seems able to find a free port after trying to bind to port 0.  Other than that, it's perfect.

Comment (yhwen) [reply nvflare/app_common/pt/pt_multi_process_executor.py:37]: bind() will find the open port. listen() is to set the connection queue. May not be absolutely necessary, but a good practice.

**Review by yanchengnv — APPROVED** [review]

File: nvflare/apis/utils/common_utils.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import socket
+
+
+def get_open_ports(number):
```
Comment (IsaacYangSLA) [inline nvflare/apis/utils/common_utils.py:18]: Roughly when number == 100, there could be one duplicate port (>1024) in the returned list.  Other than that, this function seems safe.

**Review by IsaacYangSLA — APPROVED** [review]
Just one comment about the get_open_ports function.

**Pull request merged**